sdk/helper/tokenutil/root/encode.go

```diff
@@ -0,0 +1,48 @@
+package root
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/pgpkeys"
+	"github.com/hashicorp/vault/sdk/helper/xor"
+)
+
+var (
+	// ErrNoTokenProvided is returned when an empty string is provided as an error
+	ErrNoTokenProvided = errors.New("no token provided")
+	// ErrNoOTPorPGPKeyProvided is returned when a non-empty token is provided but not OTP or PGP key
+	ErrNoOTPorPGPKeyProvided = errors.New("no otp or pgp key provided")
+)
+
+// EncodeToken gets a token and an OTP or a PGP Key and encodes it. If no OTP or PGP key is
+// provided, an error will be returned. If an OTP is provided, it must have the same length as the token
+func EncodeToken(token, otp, pgpKey string, cleanupFunc func()) (string, error) {
```

Comment:
Everything else in this patch is really about the otp, it's only this func that knows a tiny bit about encrypting the token using PGP.  I think we should narrow the focus to just the otp part for now.

Reply:
You mean split away the PGP encryption part back into the generate root portion of the Vault module?

Reply:
Right.  Just handle one aspect of token encoding here.

Reply:
Fixed. I suppose that I should move the pgpkeys package where it was originally, right?

Reply:
Yes please.  It probably makes sense to move it as you've done, but I'd rather that be in its own PR if we do it.